spec/rubocop/cop/style/slicing_with_range_spec.rb

```diff
@@ -141,5 +187,97 @@
         ary[..-1]
       RUBY
     end
+
+    context 'when calling `[]` with a dot' do
+      it 'reports an offense and corrects for `nil..42`' do
+        expect_offense(<<~RUBY)
+          ary.[](nil..42)
+             ^^^^^^^^^^^^ Prefer `..42` over `nil..42`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary.[](..42)
+        RUBY
+      end
+
+      it 'reports an offense and corrects for `nil..42` without parens' do
+        expect_offense(<<~RUBY)
+          ary.[] nil..42
+             ^^^^^^^^^^^ Prefer `..42` over `nil..42`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary.[] ..42
+        RUBY
+      end
+    end
+
+    context 'when calling `[]` with a safe navigation' do
+      it 'reports an offense and removes the entire method call for 0..-1' do
+        expect_offense(<<~RUBY)
+          ary&.[](0..-1)
+             ^^^^^^^^^^^ Remove the useless `&.[](0..-1)`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary
+        RUBY
+      end
+
+      it 'reports an offense and removes the entire method call for 0..-1 without parens' do
+        expect_offense(<<~RUBY)
+          ary.[] 0..-1
+             ^^^^^^^^^ Remove the useless `.[] 0..-1`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary
+        RUBY
+      end
+
+      it 'reports an offense and corrects for `1..-1`' do
+        expect_offense(<<~RUBY)
+          ary.[](1..-1)
+             ^^^^^^^^^^ Prefer `1..` over `1..-1`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary.[](1..)
+        RUBY
+      end
+
+      it 'reports an offense and corrects for `1..-1` without parens' do
+        expect_offense(<<~RUBY)
+          ary.[] 1..-1
+             ^^^^^^^^^ Prefer `1..` over `1..-1`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary.[] 1..
```

Comment:
When parentheses are omitted, the semantic change if there is continuation line:

```ruby
ary.[] 1..
next_line
```

In any case, this is a corner case usage, so allowing the omission of parentheses would be the simpler approach.